Assets/Scripts/Planet.cs

```diff
@@ -7,34 +7,131 @@ public class Planet : MonoBehaviour
     public float DistanceFromShip;
     private float XDisplacement, YDisplacement;
     private Rigidbody2D PlanetRigidbody;
+    private Animator PlanetAnimator;
+    private int SetLocationPasses, PlacementPriority;
+    private const int MaxPasses = 30;
 
     private void Start()
     {
+        PlanetAnimator = GetComponent<Animator>();
+        SetLocationPasses = 0;
+        PickStartLocationRandomly();
+    }
+
+    private void Update()
+    {
+        if (PlanetAnimator.GetCurrentAnimatorStateInfo(0).IsName("PlanetDestroyed"))
+        {
+            Destroy(gameObject);
+        }
+    }
+
+    private void OnCollisionEnter2D(Collision2D collision)
+    {
+        if (!collision.gameObject.CompareTag("Planet"))
+        {
+            PlanetAnimator.SetBool("PlanetHit", true);
+            Debug.Log("Game Over: A planet was destroyed by your laser.");
+        }
+        if (collision.gameObject.CompareTag("Planet"))
+        {
+            RelocateOnePlanet(collision.gameObject);
+        }
+    }
+
+    // if planets managed to get placed on top of one another twice in a row, they wouldn't reposition again without this
+    private void OnCollisionStay2D(Collision2D collision)
+    {
+        if (collision.gameObject.CompareTag("Planet"))
+        {
+            RelocateOnePlanet(collision.gameObject);
+        }
+    }
+
+    // if the planet spawned on another planet, move the lower priority planet clockwise a bit
+    public void PickStartLocationUsingPrevious()
+    {
+        // why does this line need to be here now?
         PlanetRigidbody = GetComponent<Rigidbody2D>();
```

Comment:
This line should be in the Start method.

Reply:
I tried moving it to Start, but if I don't set the Rigidbody2D inside the function I intend to use it in, I get a NullReferenceException for some reason, so we'll have to keep it here for now.